spec/bundler/ui/shell_spec.rb

```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+require "spec_helper"
+
+RSpec.describe Bundler::UI::Shell do
+  subject { described_class.new }
+
+  before { subject.level = "debug" }
+
+  describe "#info" do
+    before { subject.level = "info" }
+    it "prints to stdout" do
+      expect { subject.info("info") }.to output("info\n").to_stdout
+    end
+  end
+
+  describe "#confirm" do
+    before { subject.level = "confirm" }
+    it "prints to stdout" do
+      expect { subject.confirm("confirm") }.to output("confirm\n").to_stdout
+    end
+  end
+
+  describe "#warn" do
+    before { subject.level = "warn" }
+    it "prints to stdout" do
+      expect { subject.warn("warning") }.to output("warning\n").to_stdout
+    end
+  end
+
+  describe "#debug" do
+    it "prints to stdout" do
+      expect { subject.debug("debug") }.to output("debug\n").to_stdout
+    end
+  end
+
+  describe "#error" do
+    before { subject.level = "error" }
+    it "prints to stdout" do
+      expect { subject.error("error!!!") }.to output("error!!!\n").to_stdout
+    end
+
+    context "when stderr flag is enabled" do
+      before { bundle "config error_on_stderr true" }
+      it "prints to stderr" do
+        expect { subject.error("error!!!") }.to output("\e[31merror!!!\e[0m").to_stderr
```

Comment:
Looking into this more and looked at how `rspec-expectations` is capturing the output of `$stderr` and `$stdout` and there is some weird behaviour happening when i try and follow the same method:


```
b = StringIO.new
$stderr = b
$stderr.print "\e[31merror!!!\e[0m\n"
$stderr.flush
puts " stderr: #{b.string.inspect}"

a = StringIO.new
$stdout = a
$stdout.print "\e[31merror!!!\e[0m\n"
$stdout.flush
puts " stdout: #{a.string.inspect}"
```

```
› ruby test.rb
 stderr: "\e[31merror!!!\e[0m\n"
```

I can't see the output from `$stdout` anywhere 🤔